cpp/src/arrow/csv/options.h

```diff
@@ -170,6 +170,19 @@ struct ARROW_EXPORT ReadOptions {
   Status Validate() const;
 };
 
+/// \brief Quoting style for CSV writing
+enum class ARROW_EXPORT QuotingStyle {
+  /// Only enclose values in quotes which need them, because their CSV rendering can
+  /// contain quotes itself (e.g. strings or binary values)
+  Needed,
+  /// Enclose all valid values in quotes. Nulls are not quoted.
+  AllValid,
```

Comment:
does this lose roundtripping information for numbers?  Should that be documented?